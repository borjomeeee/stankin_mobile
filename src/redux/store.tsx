import {combineReducers, applyMiddleware, createStore} from 'redux';
import createSagaMiddleware from 'redux-saga';

import AsyncStorage from '@react-native-community/async-storage';
import {persistStore, persistReducer, createTransform} from 'redux-persist';

import AppReducer from '../reducers/App.reducer';
import UserReducer from '../reducers/User.reducer';
import SheduleReducer from '../reducers/Shedule.reducer';
import NotesReducer from '../reducers/Notes.reducer';

import RootSaga from '../saga/Root.saga';

import User from '../models/User.model';
import {INote} from '../models/Note.model';
import {ILesson} from '../models/Lesson.model';

// Initial state
export const initialState = {
  app: {
    version: '1.0.0',
    lastUpdateSchedule: new Date().getDate(),

    isLoading: false,
  },

  user: new User(),
  schedule: new Map<number, ILesson[]>(),
  notes: new Map<number, INote[]>(),
};

export type IInitialState = typeof initialState;
export type IAppInitialState = typeof initialState.app;
export type IUserInitialState = typeof initialState.user;
export type IScheduleInitialState = typeof initialState.schedule;
export type INotesInitialState = typeof initialState.notes;

// export type IShedulePersistedInitialState = [number, ILesson[]][];
// export type INotesPersistedInitialState = [number, INote[]][];

const transformMapsState = createTransform(
  (state: Map<number, any[]>) => {
    return JSON.stringify(Array.from(state || new Map<number, any[]>()));
  },
  (state: string) => {
    return new Map<number, ILesson[]>(
      JSON.parse(state.length > 0 ? state : '[]'),
    );
  },
  {whitelist: ['schedule', 'notes']},
);

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  transforms: [transformMapsState],
};
const reducers = combineReducers({
  app: AppReducer,
  user: UserReducer,
  schedule: SheduleReducer,
  notes: NotesReducer,
});

const persistedReducer = persistReducer(persistConfig, reducers);

const sagaMiddleware = createSagaMiddleware();
const store = createStore(persistedReducer, applyMiddleware(sagaMiddleware));

export default {store, persistor: persistStore(store)};

sagaMiddleware.run(RootSaga);
