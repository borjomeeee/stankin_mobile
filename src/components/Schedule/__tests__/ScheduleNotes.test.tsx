import React from 'react';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';

import ScheduleNotesComponent from '../ScheduleNotes.component';

import {initialState, reducers} from '../../../redux/store';
import {INote} from '../../../models/Note.model';

describe('Schedule notes component tests', () => {
  const currDate = new Date();

  initialState.notes.set(currDate.getTime(), [
    ...(initialState.notes.get(currDate.getTime()) || []),
    {
      id: '1',
      text: 'Написать реферат',
      subject: 'Технологии программирования',
      date: currDate,
      isChecked: false,
    },
    {
      id: '2',
      text: 'Написать эссе',
      subject: 'Технологии программирования',
      date: currDate,
      isChecked: false,
    },
    {
      id: '3',
      text: 'Нарисовать рисунок',
      subject: 'Технологии программирования',
      date: currDate,
      isChecked: true,
    },
  ] as INote[]);

  const store = createStore(reducers);

  it('Schedule notes render correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ScheduleNotesComponent currDate={new Date()} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
