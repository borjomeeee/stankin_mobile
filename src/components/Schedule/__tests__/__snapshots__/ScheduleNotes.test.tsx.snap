// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule notes component tests Schedule notes render correctly 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Inter-Bold",
          "fontSize": 18,
          "marginBottom": 10,
          "marginTop": 20,
        },
      ]
    }
  >
    Дедлайны
  </Text>
  <View
    style={
      Array [
        Object {
          "marginTop": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#F0F0F0",
            "borderStyle": "solid",
            "borderWidth": 1,
            "justifyContent": "center",
            "marginBottom": 15,
            "paddingBottom": 15,
            "paddingLeft": 60,
            "paddingRight": 60,
            "paddingTop": 15,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#C4C4C4",
              "fontFamily": "Inter-Regular",
              "fontSize": 16,
              "textAlign": "center",
            },
          ]
        }
      >
        На текущую дату у вас нет ни одного дедлайна
      </Text>
    </View>
  </View>
</View>
`;
